src/rapidfuzz/process_cpp_impl.pyx

```diff
@@ -1208,6 +1208,7 @@ def extract(query, choices, *, scorer=WRatio, processor=None, limit=5, score_cut
     cdef RF_Scorer* scorer_context = NULL
     cdef RF_ScorerFlags scorer_flags
     cdef int64_t c_limit
+    cdef int64_t choices_len = <int64_t>len(choices)
```

Comment:
This will crash when choices is a generator. So this would need to be handled inside the try/except